ehrql/query_language.py

```diff
@@ -68,14 +74,24 @@ def define_population(self, population_condition):
         validate_population_definition(population_condition._qm_node)
         self.variables["population"] = population_condition
 
+    def configure_dummy_dataset(self, *, population_size):
+        """
+        Configure the dummy dataset.
+
+        ```py
+        dataset.configure_dummy_dataset(population_size=10000)
+        ```
+        """
+        self.dummy_dataset_config.population_size = population_size
+
     def __setattr__(self, name, value):
         if name == "population":
             raise AttributeError(
                 "Cannot set variable 'population'; use define_population() instead"
             )
         if name in self.variables:
             raise AttributeError(f"'{name}' is already set and cannot be reassigned")
-        if name in ("patient_id", "variables"):
+        if name in ("patient_id", "variables", "dummy_dataset_config"):
```

Comment:
For the sake of completeness, might be worth adding this as a test case here:
https://github.com/opensafely-core/ehrql/blob/c3a3417a25cf96d66bbb2e7fb045d27bb79a8606/tests/unit/test_query_language.py#L121-L131